src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                    UtilityRoutines::MakeUPPERCase(fields.at("water_temperature_curve_input_variable").get<std::string>());
+                thisPLHP.waterTempCurveInputVar =
+                    static_cast<WaterTempCurveVar>(getEnumerationValue(WaterTempCurveVarNamesUC, waterTempCurveInputVar));
+                if (thisPLHP.waterTempCurveInputVar == WaterTempCurveVar::Invalid) {
+                    thisPLHP.waterTempCurveInputVar = WaterTempCurveVar::EnteringCondenser;
+                    // 2022-08-08: give warning for resetting
+                }
+
+                // A11 normalized_capacity_function_of_temperature_curve_name
+                auto &capFtName = fields.at("normalized_capacity_function_of_temperature_curve_name");
+                thisPLHP.capFuncTempCurveIndex = Curve::GetCurveIndex(state, UtilityRoutines::MakeUPPERCase(capFtName.get<std::string>()));
+                if (thisPLHP.capFuncTempCurveIndex == 0) {
+                    ShowSevereError(state,
+                                    fmt::format("Invalid curve name for EIR PLFFHP (name={}; entered curve name: {}",
+                                                thisPLHP.name,
+                                                capFtName.get<std::string>()));
+                    errorsFound = true;
+                }
+
+                // A12 fuel_energy_input_ratio_function_of_temperature_curve_name
+                auto &eirFtName = fields.at("fuel_energy_input_ratio_function_of_temperature_curve_name");
+                thisPLHP.powerRatioFuncTempCurveIndex = Curve::GetCurveIndex(state, UtilityRoutines::MakeUPPERCase(eirFtName.get<std::string>()));
+                if (thisPLHP.capFuncTempCurveIndex == 0) {
+                    ShowSevereError(state,
+                                    fmt::format("Invalid curve name for EIR PLFFHP (name={}; entered curve name: {}",
+                                                thisPLHP.name,
+                                                eirFtName.get<std::string>()));
+                    errorsFound = true;
+                }
+                // A13 fuel_energy_input_ratio_function_of_plr_curve_name
+                auto &eirFplrName = fields.at("fuel_energy_input_ratio_function_of_plr_curve_name");
+                thisPLHP.powerRatioFuncPLRCurveIndex = Curve::GetCurveIndex(state, UtilityRoutines::MakeUPPERCase(eirFplrName.get<std::string>()));
+                if (thisPLHP.capFuncTempCurveIndex == 0) {
+                    ShowSevereError(state,
+                                    fmt::format("Invalid curve name for EIR PLFFHP (name={}; entered curve name: {}",
+                                                thisPLHP.name,
+                                                eirFplrName.get<std::string>()));
+                    errorsFound = true;
+                }
+
+                // N6 min PLR
+                if (fields.find("minimum_part_load_ratio") != fields.end()) {
```

Comment:
Same comment as above, don't search for `minimum_part_load_ratio` twice.

Reply:
Addressed in earlier changes.